editor/state.js

```diff
@@ -136,6 +156,26 @@ export const blocks = combineUndoableReducers( {
 }, { resetTypes: [ 'RESET_BLOCKS' ] } );
 
 /**
+ * Reducer returning the last-known state of the current post, in the format
+ * returned by the WP REST API.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Dispatched action
+ * @return {Object}        Updated state
+ */
+export function currentPost( state = {}, action ) {
```

Comment:
Per your previous comment, still unclear the value or complications we'll encounter with storing the full post object, vs. just the immediate need of `currentPostId` (a [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) consideration).